src/main/java/org/wso2/carbon/module/swiftiso20022/JsonToMT940Transformer.java

```diff
@@ -0,0 +1,210 @@
+     */
+    private ErrorModel validateRequestPayload(RequestPayloadModel requestPayload) {
+
+        if (StringUtils.isBlank(requestPayload.getBlock1())) {
+            return new ErrorModel(ConnectorConstants.ERROR_T13, ConnectorConstants.ERROR_BLOCK1_INVALID);
+        }
+
+        if (StringUtils.isBlank(requestPayload.getBlock2())) {
+            return new ErrorModel(ConnectorConstants.ERROR_T13, ConnectorConstants.ERROR_BLOCK2_INVALID);
+        }
+
+        if (!JsonToMt940Utils.validateAccountNumber(requestPayload)) {
+            return new ErrorModel(ConnectorConstants.ERROR_M50, ConnectorConstants.ERROR_ACC_NO_INVALID);
+        }
+
+        if (!JsonToMt940Utils.validateReference(requestPayload.getReference())) {
+            return new ErrorModel(ConnectorConstants.ERROR_M50, ConnectorConstants.ERROR_REF_INVALID);
+        }
+
+        if (!JsonToMt940Utils.validateSequenceNumber(requestPayload)) {
+            return new ErrorModel(ConnectorConstants.ERROR_M50, ConnectorConstants.ERROR_SEQ_NO_INVALID);
+        }
+
+        if (requestPayload.getOpeningBalanceDetails() == null) {
+            return new ErrorModel(ConnectorConstants.ERROR_T13,
+                    String.format(ConnectorConstants.ERROR_BALANCE_MISSING, ConnectorConstants.OPENING_BALANCE));
+        }
+
+        ErrorModel errorModel = JsonToMt940Utils.validateBalanceDetails(requestPayload.getOpeningBalanceDetails(),
+                ConnectorConstants.OPENING_BALANCE);
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        if (requestPayload.getClosingBalanceDetails() == null) {
+            return new ErrorModel(ConnectorConstants.ERROR_T13,
+                    String.format(ConnectorConstants.ERROR_BALANCE_MISSING, ConnectorConstants.CLOSING_BALANCE));
+        }
+
+        errorModel = JsonToMt940Utils.validateBalanceDetails(requestPayload.getClosingBalanceDetails(),
+                ConnectorConstants.CLOSING_BALANCE);
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        if (requestPayload.getClosingAvailableBalanceDetails() != null) {
+            errorModel = JsonToMt940Utils.validateBalanceDetails(requestPayload.getClosingAvailableBalanceDetails(),
+                    ConnectorConstants.CLOSING_AVAIL_BALANCE);
+            if (errorModel.isError()) {
+                return errorModel;
+            }
+        }
+
+        if (requestPayload.getForwardAvailableBalanceDetails() != null) {
+            errorModel = JsonToMt940Utils.validateBalanceDetails(requestPayload.getForwardAvailableBalanceDetails(),
+                    ConnectorConstants.FORWARD_CLOSING_AVAIL_BALANCE);
+            if (errorModel.isError()) {
+                return errorModel;
+            }
+        }
+
+        ErrorModel error = validateTransactionDetails(requestPayload.getTransactions());
+        if (error.isError()) {
+            return error;
+        }
+
+        error.setIsError(false);
```

Comment:
Do we need to set is error false here? For non error flows this value is already set by https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/files#diff-0cf0c4360b95e74c7f840a4544aa46af6e986fd53932e1f49b4f9cf8e1cb503cR207

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/b42a3d413ebea8b6bb28d21d52e7f729488b979f